pandas/io/formats/format.py

```diff
@@ -1094,7 +1097,10 @@ def format_values_with(float_format):
                                            for val in values.ravel()[imask]])
 
             if self.fixed_width:
-                return _trim_zeros(values, na_rep)
+                if is_complex:
```

Comment:
you can do the type check right here instead of above

Reply:
The problem with that is at that point the `values` object has been reassigned as `            values = np.array(values, dtype='object')` and is no longer a complex type but an array with the complex numbers as strings.